src/globus_sdk/experimental/auth_requirements_error/_variants.py

```diff
@@ -0,0 +1,312 @@
+        """
+        Return a GlobusAuthRequirementsError representing this error.
+
+        Normalizes the required_scope field to a list and uses the description
+        to set the session message.
+        """
+        return GlobusAuthRequirementsError(
+            code=self.code,
+            authorization_parameters=GlobusAuthorizationParameters(
+                required_scopes=[self.required_scope],
+                session_message=self.extra_fields.get("description"),
+                extra=self.extra_fields.get("authorization_parameters"),
+            ),
+            extra={
+                k: v
+                for k, v in self.extra_fields.items()
+                if k != "authorization_parameters"
+            },
+        )
+
+
+class LegacyAuthorizationParameters:
+    """
+    An Authorization Parameters object that describes all known variants in use by
+    Globus services.
+    """
+
+    session_message: str | None
+    session_required_identities: list[str] | None
+    session_required_policies: str | list[str] | None
+    session_required_single_domain: str | list[str] | None
+    session_required_mfa: bool | None
+    # Declared here for compatibility with mixed legacy payloads
+    required_scopes: list[str] | None
+    extra_fields: dict[str, t.Any]
+
+    DEFAULT_CODE = "AuthorizationRequired"
+
+    SUPPORTED_FIELDS = {
+        "session_message": validators.OptionalString,
+        "session_required_identities": validators.OptionalListOfStrings,
+        "session_required_policies": (
+            validators.OptionalListOfStringsOrCommaDelimitedStrings
+        ),
+        "session_required_single_domain": (
+            validators.OptionalListOfStringsOrCommaDelimitedStrings
+        ),
+        "session_required_mfa": validators.OptionalBool,
+    }
+
+    def __init__(
+        self,
+        *,
+        session_message: str | None = None,
+        session_required_identities: list[str] | None = None,
+        session_required_policies: str | list[str] | None = None,
+        session_required_single_domain: str | list[str] | None = None,
+        session_required_mfa: bool | None = None,
+        extra: dict[str, t.Any] | None = None,
+    ):  # pylint: disable=unused-argument
+        # Validate and assign supported fields
+        for field_name, validator in self.SUPPORTED_FIELDS.items():
+            try:
+                field_value = validator(locals()[field_name])
+            except ValueError as e:
+                raise ValueError(f"Error validating field '{field_name}': {e}") from e
+
+            setattr(self, field_name, field_value)
+
+        # Retain any additional fields
+        self.extra_fields = extra or {}
+
+        # Enforce that the error contains at least one of the fields we expect
+        if not any(
+            (getattr(self, field_name) is not None)
+            for field_name in self.SUPPORTED_FIELDS.keys()
+        ):
```

Comment:
Could I recommend this, to avoid double negatives?

```suggestion
        if all(
            getattr(self, field_name) is None
            for field_name in self.SUPPORTED_FIELDS
        ):
```

Reply:
Nice catch! 👍